modules/redisearch.ts

```diff
@@ -111,84 +116,93 @@ export class Redisearch extends Module {
-                args = args.concat(['FILTER', parameters.filter.field, parameters.filter.min.toString(), parameters.filter.max.toString()])
+            if(parameters.filter !== undefined) {
+                for (const filterItem of parameters.filter) {
+                    args = args.concat(['FILTER', filterItem.field, `${filterItem.min}`, `${filterItem.max}`])
+                }
+            }
             if(parameters.geoFilter !== undefined)
                 args = args.concat([
                     'GEOFILTER',
                     parameters.geoFilter.field,
-                    parameters.geoFilter.lon.toString(),
-                    parameters.geoFilter.lat.toString(),
-                    parameters.geoFilter.radius.toString(),
+                    `${parameters.geoFilter.lon}`,
+                    `${parameters.geoFilter.lat}`,
+                    `${parameters.geoFilter.radius}`,
                     parameters.geoFilter.measurement
                 ])
             if(parameters.inKeys !== undefined)
-                args = args.concat(['INKEYS', parameters.inKeys.num.toString(), parameters.inKeys.field])
+                args = args.concat(['INKEYS', `${parameters.inKeys.length}`].concat(parameters.inKeys.map(inKeysItem => `${inKeysItem}`)))
             if(parameters.inFields !== undefined)
-                args = args.concat(['INFIELDS', parameters.inFields.num.toString(), parameters.inFields.field])
+                args = args.concat(['INFIELDS', `${parameters.inFields.length}`].concat(parameters.inFields.map(inFieldItem => `${inFieldItem}`)))
             if(parameters.return !== undefined) {
-                args.push('RETURN');
-                if(parameters.return.num)
-                    args.push(parameters.return.num.toString());
-                if(parameters.return.fields)    
-                    parameters.return.fields.forEach(field => {
-                        args.push(field.name)
-                        if(field.as)
-                            args = args.concat(['AS', field.as])
-                    })
+                args.push('RETURN')
+                let itemCount = 0
+                let tempList = []
+                for (const returnItem of parameters.return) {
+                    if(typeof returnItem === "string") {
+                        //Strings can be pushed directly
+                        tempList.push(returnItem)
+                        itemCount++
+                    } else {
+                        //Objects need to be converted to strings
+                        tempList.push(returnItem.field)
+                        itemCount++
+                        if(returnItem.as !== undefined) {
+                            tempList = tempList.concat(["AS", returnItem.as])
+                            itemCount += 2
+                        }
+                    }
+                }
+                args.push(`${itemCount}`)
+                args = args.concat(tempList)
```

Comment:
As much as this easier to use you are blocking the ability of users to negatively test the Search module.
Please adjust the 'return' key object to num as number and make the fields as a list. (In the type)
i.e.
return: {
    num: number,
    fields: {
            name: string,
            as?: string
    }
}
Maybe, if you feel like the `num` key is redundant in a most cases, you could make it optional to pass else it will count it the you did.